### ISSUE TITLE
divide loss_sam1 by (loss_sam1/loss1).detach()

### ISSUE THREAD

**Issue by Lilyo** [dwHou/EMDC-PyTorch#5]
Hi @dwHou ,

Thanks for your great work again! One more question, I'm curious about why you divide loss_sam1 by `(loss_sam1/loss1).detach()`

Does it equal to loss = loss + loss_sam1 * (loss1/loss_sam1) in math? and why?

https://github.com/dwHou/EMDC-PyTorch/blob/07867c78c4930b78f76ce78c4d1ac3868c0a614a/train_model.py#L127-L133

Looking forward to your reply, thanks.

**Comment by dwHou** [comment]
Hi @Lilyo, 
Yes, it is equivalent to loss = loss + loss_sam1*(loss1/loss_sam1).detach() . The purpose is to make loss_sam1 and loss_sam2 the same scale, even the same value here.  
This way, the global branch1 and local branch of their back-propagation are learned with a balanced step. Avoid model mismatch in the middle of training, which will lead to unstable training of the fusion module.

**Comment by Lilyo** [comment]
Hi @dwHou , 
It’s pretty clear! Thanks!
